doc/ReferenceManual.rst

```diff
@@ -161,7 +161,7 @@ Misc
     Slot("__new__", "tp$new", "new"),
     Slot("__init__", "tp$init", "init"),
     Slot("__str__", "tp$print", "print"),
-    Slot("__repr__", "tp$repr", "repr",
+    Slot("__repr__", "$r", "repr",
```

Comment:
Changes here are a reflection of what exists. $r  for example is our equivalent of cpythons tp$repr. So I update this doc to reflect the code base as it currently stands. 

I replaced nb$nonzero with nb$bool since that’s a python2ism but we still support __nonzero__ as a dunder method in py2 mode it will just get the slot function nb$bool. And this switch is handled in the switch to py2 env function. 


I’ve tried to move more towards cpython hopefully since we now don’t have mp$del_subscript for example. Which was skulpt only. 


Curiously Cpython doesn’t have reflected slots like skulpt does. But the way cpython handles reflected number slots wouldn’t work the same way in skulpt. 

I’ve tried to document all the slots we support in the jsdocs. Or as comments in slotdefs.js.

Reply:
Additional things. 

nb$divide is depreciated in cpython3
Because py2 has both nb$divide and nb$true_divide. 

I figured nb$divide is nicer and we can use that since we were already using it over nb$true_divide. But happy to move this in-line with cpython3. Its an easy change I think. 

nb$lng has a few versions in the code base but since nb$int_ matches its constructor name I figured nb$long should too

sq$ass_slice is not in cpython3 so I removed this from the codebase. 

I also use tp$as_number which is a cpython slot.  

Cpython has two slots tp$as_sequence and tp_as_mapping. 

I combine these as tp$as_sequence_or_mapping since skulpt handles these synonymously.

Reply:
To resolve this I have updated this rst with the current slots in the codebase. 
The document was not previously up to date with master.
The names should now hopefully be correct at least.

Reply:
thank you!